### ISSUE TITLE
sile's autoconf files are outdated and needs update

### ISSUE THREAD

**Issue by moodyhunter** [sile-typesetter/sile#1423]
Hello, I have encountered the following error message when porting sile to the RISC-V platform, following the build script here[2]:

```
This script, last modified 2014-03-23, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (build-aux/config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.
```

(original log can be seen here[1].)

Please consider updating the Autoconf version.
Thanks :)

[1]: https://archriscv.felixc.at/.status/log.htm?url=logs/sile/sile-0.13.0-1.log
[2]: https://github.com/archlinux/svntogit-community/blob/packages/sile/trunk/PKGBUILD#L59

**Comment by alerque** [comment]
Hmmm, happy to look into this. I'm wondering though if this is a function of what is in the repo as far as our autotools implementation goes or if it is the generated scripts that get added as part of the `make dist` process that generates the source tarball.

Are you starting from a Git clone or the source tarball? Have you tried running `autoreconf --install --force` to update the scripts yourself locally? It would be nice to confirm whether that does the trick to recognize your platform.

**Comment by moodyhunter** [comment]
Hi, we are currently using the tarball from the release, and yes `autoreconf -fiv` does the trick and resolves the problem :)

**Comment by alerque** [comment]
Great then thanks for the info...

Out of curiosity, what hardware are you running this on?

**Comment by moodyhunter** [comment]
However, another error occurs after a successful build, when I'm running `make check` on a RISC-V vm:

```
[root@host sile-0.13.0]# make check
output=$(mktemp -t selfcheck-XXXXXX.pdf)
trap 'rm -f $output' EXIT HUP TERM
echo "<sile>foo</sile>" | ./sile -o $output -
/usr/bin/pdfinfo $output | /usr/bin/grep "SILE v0.13.0"
SILE v0.13.0 (Lua 5.4)
</dev/stdin>

Error detected:
        /usr/share/lua/5.4/cassowary/init.lua:488: attempt to concatenate a nil value (field 'value')
Syntax Error: Document stream is empty
make: *** [Makefile:1400: selfcheck] Error 1
```

It's not occurring on an x86 laptop:

```
src/sile-0.13.0 community/packages/sile> make check
touch .built-subdirs
output=$(mktemp -t selfcheck-XXXXXX.pdf)
trap 'rm -f $output' EXIT HUP TERM
echo "<sile>foo</sile>" | ./sile -o $output -
/usr/bin/pdfinfo $output | /usr/bin/grep "SILE v0.13.0"
SILE v0.13.0 (Lua 5.4)
</dev/stdin>

! Error loading localizations en: no localizations for this language at /dev/stdin: in \vfill
[1]
Producer:        SILE v0.13.0 (Lua 5.4)
......
```

**Comment by moodyhunter** [comment]
> Great then thanks for the info...
> 
> Out of curiosity, what hardware are you running this on?

[We](https://archriscv.felixc.at/) currently have RISC-V development boards (sifive unmatched) and some qemu virtual machines,

**Comment by alerque** [comment]
So this comes down to the fact that we're building source tarballs in GitHub Actions CI runners. At present the default host environment is the Ubuntu Focal 20.04 LTS which comes with automake 1.16.1 (autotools-dev package from 20180224). It looks like they have a beta image for the new LTS Jammy 22.04 available now which has automake 1.16.5 (autotools-dev from 20220109).

The three things I can think of to fix this for you are:

1. Try generating a source tarball from Jammy and see if that is new enough.
2. Manually upgrade autotools stuff on the CI runner host before use.
3. Just document that some newer platforms may need to run `autoreconf`.

Lets try those in order. I'll spin up a test build...

**Comment by alerque** [comment]
Alright we have a source tarball generated from [this test run](https://github.com/sile-typesetter/sile/actions/runs/2480435947) ([sile-0.13.0.r20-g81fee85.zip.zip ](https://github.com/sile-typesetter/sile/suites/6891646476/artifacts/267145902)). These non-release artifacts are super awkward zips of the dist zip, but once you peal off both wrappers it should be the same thing in the tar'ed release artifacts.

@moodyhunter Would you be able to confirm these are built with a new enough autotools build to support your RISC-V stuff out of the box? If not I'll try something else. I don't actually know what autotools release got the new platform support, only that this moves the needle up about 2 years.

**Comment by moodyhunter** [comment]
hm unfortunately no, it seems the error is emitted from `libtexpdf`?

```
...
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating sile.1
config.status: creating core/version.lua
config.status: creating sile
config.status: executing depfiles commands
config.status: executing libtool commands
=== configuring in libtexpdf (/root/sile-0.13.0.r20-g81fee85/libtexpdf)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  '--with-system-luarocks' --cache-file=/dev/null --srcdir=.
checking build system type... ./build-aux/config.guess: unable to guess system type

This script, last modified 2014-03-23, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./build-aux/config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2014-03-23
```

**Comment by alerque** [comment]
Thanks, I'll look into that. The libtexpdf module build system is kind of spliced in in the middle in a weird way, it probably needs some love. It *really* should be an external dependency anyway. I may not extract it in this release cycle but we can probably bump the scripts over there in a patch release.

**Comment by moodyhunter** [comment]
Thanks for the fast reply, it's no hurry, and I'll be happy to test anything if required.

**Comment by alerque** [comment]
The `libtexpdf` library repository (that we pull in as a Git submodule) has some absolute cruft inherited from it's original home embedded in dvipdfm. This includes *generated* stuff that should be created at release time not tracked in the repository in the state it was generated 8 years ago.

I'm going to merge the other bumps anyway because it turns out the newer Ubuntu build env cleans up some other issues for us, but I'll keep this open to track this not being fixed yet.

I'll ping you when there is something to test, thanks!

**Comment by moodyhunter** [comment]
Sounds good!

**Comment by alerque** [comment]
Alright @moodyhunter I had another go at this. It looks like there was just a bunch of cruft in the libtexpdf repo where generated scripts were being tracked in the repository. I deleted them entirely and now when we generate the source distribution tarballs we get fresh scripts.

Can you have a go at the artifacts from [this CI run]()?

**Comment by moodyhunter** [comment]
yes! fortunately, it now compiles, 
but I get the following error when running `make check`:

```
==> Starting check()...
touch .built-subdirs
output=$(mktemp -t selfcheck-XXXXXX.pdf)
trap 'rm -f $output' EXIT HUP TERM
echo "<sile>foo</sile>" | ./sile -o $output -
/usr/bin/pdfinfo $output | /usr/bin/grep "SILE v0.13.0.r29-g4cdf0b6"
SILE v0.13.0.r29-g4cdf0b6 (Lua 5.4)
</dev/stdin>

Error detected:
        /usr/share/lua/5.4/cassowary/init.lua:488: attempt to concatenate a nil value (field 'value')
Syntax Error: Document stream is empty
make: *** [Makefile:1399: selfcheck] Error 1
==> ERROR: A failure occurred in check().
```

**Comment by alerque** [comment]
@moodyhunter That particular error is being thrown by the Cassowary library installed as a Lua Rock (either vendored with SILE if you used `./configure --without-system-luarocks` if you went that route or from your system if you did the opposite). The bug was fixed in v2.3.2 specifically before the release of SILE v0.13.0 which started triggering it. To make that error go away you need to update Cassowary however it is being supplied on your system. I assume this is not a clean source build using `--without-system-luarocks` or it would already have installed a local copy and v2.3.2.

**Comment by alerque** [comment]
Thanks for checking on the RISC-V platform support. I've merged that PR and the next release will be expected to work out of the box.
